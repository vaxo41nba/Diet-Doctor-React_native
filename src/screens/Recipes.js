import React from 'react';
import {
  Dimensions,
  Image,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import images from '../constants/images';

const Item = ({type}) => (
  <TouchableOpacity style={styles.type}>
    <Image source={images[type]} style={styles.image} />
    <Text style={styles.typeText}>{type}</Text>
  </TouchableOpacity>
);

const {width} = Dimensions.get('window');
const imageWidth = width / 2 - 15;
const imageHeight = imageWidth * 0.6;

export default function Recipes() {
  const popular = ['Chicken', 'Keto', 'Cake', 'Bread', 'Salad', 'Pizza'];

  return (
    <>
      <View style={{height: 110}} />

      <View style={styles.block}>
        <ScrollView>
          <Text>Popular</Text>

          <View style={styles.typeWrapper}>
            {popular.map(p => (
              <Item key={p} type={p} />
            ))}
          </View>
        </ScrollView>
      </View>
    </>
  );
}

const styles = StyleSheet.create({
  block: {
    flex: 1,
    backgroundColor: 'white',
    borderTopLeftRadius: 40,
    paddingHorizontal: 10,
    paddingTop: 40,
  },
  input: {
    flexDirection: 'row',
    height: 50,
    width: '90%',
    backgroundColor: 'white',
    paddingHorizontal: 15,
    borderRadius: 10,
    alignSelf: 'center',
    alignItems: 'center',
    marginVertical: 30,

    shadowColor: '#000',
    shadowOffset: {
      width: 4,
      height: 4,
    },
    shadowOpacity: 0.3,
    shadowRadius: 3.0,
    elevation: 5,
  },
  filter: {
    borderWidth: 1,
    width: 40,
    height: 40,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    width: 25,
    height: 25,
    resizeMode: 'contain',
  },
  image: {
    height: imageHeight,
    width: imageWidth,
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
  },
  typeWrapper: {
    flexDirection: 'row',
    gap: 10,
    flexWrap: 'wrap',
    marginTop: 15,
  },
  type: {
    width: imageWidth,
    height: imageHeight + 30,
    alignItems: 'center',
  },
});
