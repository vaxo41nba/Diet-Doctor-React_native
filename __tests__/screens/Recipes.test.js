import React from 'react';
import renderer from 'react-test-renderer';
import Recipes from '../../src/screens/Recipes';

test('renders correctly', () => {
  const tree = renderer.create(<Recipes />).toJSON();
  expect(tree).toMatchSnapshot();
});
