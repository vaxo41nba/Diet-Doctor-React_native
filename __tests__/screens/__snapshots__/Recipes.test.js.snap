// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <View
    style={
      {
        "height": 110,
      }
    }
  />,
  <View
    style={
      {
        "backgroundColor": "white",
        "borderTopLeftRadius": 40,
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingTop": 40,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text>
          Popular
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "gap": 10,
              "marginTop": 15,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "height": 246,
                "opacity": 1,
                "width": 360,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/Chicken.png",
                }
              }
              style={
                {
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                  "height": 216,
                  "width": 360,
                }
              }
            />
            <Text>
              Chicken
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "height": 246,
                "opacity": 1,
                "width": 360,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/Keto.png",
                }
              }
              style={
                {
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                  "height": 216,
                  "width": 360,
                }
              }
            />
            <Text>
              Keto
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "height": 246,
                "opacity": 1,
                "width": 360,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/Cake.png",
                }
              }
              style={
                {
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                  "height": 216,
                  "width": 360,
                }
              }
            />
            <Text>
              Cake
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "height": 246,
                "opacity": 1,
                "width": 360,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/Bread.png",
                }
              }
              style={
                {
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                  "height": 216,
                  "width": 360,
                }
              }
            />
            <Text>
              Bread
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "height": 246,
                "opacity": 1,
                "width": 360,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/Salad.png",
                }
              }
              style={
                {
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                  "height": 216,
                  "width": 360,
                }
              }
            />
            <Text>
              Salad
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "height": 246,
                "opacity": 1,
                "width": 360,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/Pizza.png",
                }
              }
              style={
                {
                  "borderTopLeftRadius": 15,
                  "borderTopRightRadius": 15,
                  "height": 216,
                  "width": 360,
                }
              }
            />
            <Text>
              Pizza
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;
